### DISCUSSION TITLE
whatever the file i click, i wish it to be downloaded

### DISCUSSION THREAD

**Discussion by Breizyann** [DirectoryLister/DirectoryLister#1135] [Help & Support] [answered]
Hello,

My simple wish: whatever the file i click, i wish it to be downloaded immediatly and not executed or opened.

ex: if i a single clik a .jpg, a .mp3 or a .mp4, such file open on my computer instead of starting download immediatly as i wish.
With some other files extensions it already does so and i would like it come the default for every file, whatever the file extension.

If a php expert/specialist can provide me an php tweaking option to have this feature, in advance thank for sharing.
I will then add such option to my personal config.

Kind regards,
Yann

ps: i am aware of the right clik option "save link as" but if i share a file with some my relatives or friends they are not always computer specialist

**Comment by PHLAK** [comment, marked as answer]
Directory Lister does not handle the way files are opened or downloaded directly. This is handled in combination by the web server and the user's web browser.

First and foremost ensure your server is configured to [return the correct MIME type](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Configuring_server_MIME_types) of files.

  - For Apache this is accomplished with the [`AddType` directive](https://httpd.apache.org/docs/current/mod/mod_mime.html#addtype) (part of `mod_mime`)
  - For NGINX this is handled by the [`types` configuration option](http://nginx.org/en/docs/http/ngx_http_core_module.html#types) (usually found in the [`mime.types`](https://www.nginx.com/resources/wiki/start/topics/examples/full/#mime-types) file)

Additionally, your browser may be configured to handle files of a given type in a specific fashion (i.e. open vs download). Reference your browsers support page (e.g. [Firefox](https://support.mozilla.org/en-US/kb/change-firefox-behavior-when-open-file), [Chrome](https://support.google.com/chrome/answer/95759)) for instructions on configuring this.

**Reply by Breizyann** [reply]
Great thank PHLAK for prompt answering my request.

As my server is running Apache i will have a look at the AddType directive.
Also thank for sharing Directory Lister.

Kind regards,
Yann
